Metadata/MetadataConverter.php

```diff
@@ -37,49 +37,56 @@ public function convert(ClassHierarchyMetadata $metadata)
         $definitions = array();
 
         $previous = null;
+        /** @var ClassMetadata $classMetadata */
         foreach ($metadata->classMetadata as $classMetadata) {
-            if (null === $previous && null === $classMetadata->parent) {
-                $definition = new Definition();
-            } else {
-                $definition = new DefinitionDecorator(
-                    $classMetadata->parent ?: $previous->id
-                );
-            }
+            foreach ($classMetadata->getServices() as $service) {
+                if (null === $previous && null === @$service['parent']) {
+                    $definition = new Definition();
+                } else {
+                    $definition = new DefinitionDecorator(
+                        @$service['parent'] ?: $previous->id
+                    );
+                }
 
-            $definition->setClass($classMetadata->name);
-            if (null !== $classMetadata->scope) {
-                $definition->setScope($classMetadata->scope);
-            }
-            if (null !== $classMetadata->public) {
-                $definition->setPublic($classMetadata->public);
-            }
-            if (null !== $classMetadata->abstract) {
-                $definition->setAbstract($classMetadata->abstract);
-            }
-            if (null !== $classMetadata->arguments) {
-                $definition->setArguments($classMetadata->arguments);
-            }
+                $definition->setClass($classMetadata->name);
+                if (null !== @$service['scope']) {
```

Comment:
at first I only added values to the array if they existed in AnnotationDriver ~ line 82, I will change the checks now to !empty